glean-core/android/src/main/java/mozilla/telemetry/glean/Glean.kt

```diff
@@ -75,6 +107,102 @@ open class GleanInternalAPI internal constructor () {
+
+    /**
+     * Enable or disable Glean collection and upload.
+     *
+     * Metric collection is enabled by default.
+     *
+     * When uploading is disabled, metrics aren't recorded at all and no data
+     * is uploaded.
+     *
+     * When disabling, all pending metrics, events and queued pings are cleared.
+     *
+     * When enabling, the core Glean metrics are recreated.
+     *
+     * @param enabled When true, enable metric collection.
+     */
+    fun setUploadEnabled(enabled: Boolean) {
+        LibGleanFFI.INSTANCE.glean_set_upload_enabled(handle, enabled.toByte())
+    }
+
+    /**
+     * Get whether or not Glean is allowed to record and upload data.
+     */
+    fun getUploadEnabled(): Boolean {
+        return LibGleanFFI.INSTANCE.glean_is_upload_enabled(handle) != 0.toByte()
+    }
+
+    /**
+     * Indicate that an experiment is running. Glean will then add an
+     * experiment annotation to the environment which is sent with pings. This
+     * information is not persisted between runs.
+     *
+     * @param experimentId The id of the active experiment (maximum
+     *     30 bytes)
+     * @param branch The experiment branch (maximum 30 bytes)
+     * @param extra Optional metadata to output with the ping
+     */
+    fun setExperimentActive(
+        experimentId: String,
+        branch: String,
+        extra: Map<String, String>? = null
+    ) {
+        Log.e(LOG_TAG, "setExperimentActive is a stub")
+        // TODO: stub
+        // ExperimentsStorageEngine.setExperimentActive(experimentId, branch, extra)
+    }
+
+    /**
+     * Indicate that an experiment is no longer running.
+     *
+     * @param experimentId The id of the experiment to deactivate.
+     */
+    fun setExperimentInactive(experimentId: String) {
+        Log.e(LOG_TAG, "setExperimentInactive is a stub")
+        // TODO: stub
+        // ExperimentsStorageEngine.setExperimentInactive(experimentId)
+    }
+
+    /**
+     * Tests whether an experiment is active, for testing purposes only.
+     *
+     * @param experimentId the id of the experiment to look for.
+     * @return true if the experiment is active and reported in pings, otherwise false
+     */
+    @VisibleForTesting(otherwise = VisibleForTesting.NONE)
+    fun testIsExperimentActive(experimentId: String): Boolean {
+        Log.e(LOG_TAG, "testIsExperimentActive is a stub")
+        // TODO: stub
+        // return ExperimentsStorageEngine.getSnapshot()[experimentId] != null
+        return false
+    }
+
+    /**
+    * Returns the stored data for the requested active experiment, for testing purposes only.
+    *
+    * @param experimentId the id of the experiment to look for.
+    * @return the [RecordedExperimentData] for the experiment
+    * @throws [NullPointerException] if the requested experiment is not active
+    */
+    /*
+    @VisibleForTesting(otherwise = VisibleForTesting.NONE)
+    fun testGetExperimentData(experimentId: String): RecordedExperimentData {
```

Comment:
I think we can uncomment the function and simply return a bogus `RecordedExperimentData`. Plus add the bug number TODO, log and assert.